test/acceptance/features/environment.py

```diff
@@ -55,3 +62,26 @@ def before_scenario(_context, _scenario):
     _context.bindings = dict()
     output, code = cmd.run(f'{ctx.cli} get ns default -o jsonpath="{{.metadata.name}}"')
     assert code == 0, f"Checking connection to OS cluster by getting the 'default' project failed: {output}"
+
+
+def after_scenario(context, scenario):
+    global scenario_status
+    context.scenario_status = scenario_status
+    namespace = os.getenv("TEST_NAMESPACE")
+    if delete_namespace == "alwaysdelete":
+        if scenario.status == Status.passed:
+            output, code = cmd.run(f"{ctx.cli} delete namespace {namespace}.")
+            print(f"Namespace {namespace} deleted.")
+        else:
+            print(f"Deleting namespace {namespace} skipped since the scenario has failed.")
+
+    elif (delete_namespace == "keepwhenfailed") and (context.scenario_status):
+        if scenario.status == Status.passed:
+            output, code = cmd.run(f"{ctx.cli} delete namespace {namespace}.")
+            print(f"Namespace {namespace} deleted.")
+        else:
+            scenario_status = False
```

Comment:
You are setting the global variable but not using it in any condition or decision making code :)

Reply:
since all scenarios are independent im assinging the global variable to context.scenario_status  like context.scenario_status = scenario_status.
so im using the context.scenario_status to check for the keepwhenfailed scenario.
if i do not use context then there is linting errors too.

Reply:
Right, I missed that... however, the use of this global variable is redundant and not neccessary, in that case. You can use `context.scenario_status` (or `context.all_scenarios_passed` as suggested in another comment) directly. In such case you can set the initial value `True` in the `before_all` hook function.

Reply:
or use the `Environment` class as suggested in the other comment instead.